tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal

```diff
@@ -235,3 +235,14 @@ type CustomerTableWithKTC table<CustomerDetail> key<Name>;
 
 CustomerTableWithKTC tbl4 = table key(firstname, lastname) [{name: {fname: "Sanjiva", lname: "Weerawarana"},
                 id: 13, address: "Sri Lanka"}];
+
+function variableNameFieldAsKeyField() {
+    int id = 1;
+
+    table<record {readonly int id; string name;}> key (id) _ = table [
+        {id, name: "Jo"},
```

Comment:
Should this fail? id here is rhs expression and it is int. Shouldn't we consider that as a readonly value?

Reply:
Yes, this is failing. please refer to the error assertion in https://github.com/ballerina-platform/ballerina-lang/blob/d808aeba2bfb90e2a96c84765ae75e8cc66f449f/tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/table/TableNegativeTest.java#L112

Reply:
Here the error is because `id` is not a const-expr.
From the spec,
> For every field-name in the key sequence of the inherent type every mapping-constructor-expr in the row-list must include a specific-field that has a value-expr that is a const-expr.